src/ifcopenshell-python/ifcopenshell/api/geometry/add_representation.py

```diff
@@ -543,6 +543,20 @@ def create_triangulated_face_set(self):
         else:
             items = [self.file.createIfcTriangulatedFaceSet(coordinates, None, None, i) for i in ifc_raw_items if i]
 
+#This assigns every triangle their own Cartesian Points.          
+        tempCoord_IndexList=[]
+        finalCoord_IndexList=[]
+        if self.settings["should_generate_uvs"]:
+            for x in (coord_index):
```

Comment:
One problem at least that I see is that you depend on `coord_index` which is a scoped to a for loop, but use it outside of the loop. That's for me a clear indication, that the code will likely have issues in case of multiple elements in `ifc_raw_items`. Where your code will only inspect the last element, but affect `items[0]`. I would propose to merge your code with the block starting on line `534`.

Also please run `black` (`python -m black add_representation.py`) so that the code formatting is consistent with the rest of the code base.